### ISSUE TITLE
panic handling request textDocument/documentHighlight: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by gabritto** [microsoft/typescript-go#3462]
Copied from `@typescript-bot` results on PR #3454.

Source PR: https://github.com/microsoft/typescript-go/pull/3454

This error was reported in 1 bot report section(s) across 1 affected repo(s).

Source comments:
- https://github.com/microsoft/typescript-go/pull/3454#issuecomment-4285883821

Affected repos:
- firecrawl/firecrawl

Representative raw report:

### panic handling request textDocument/documentHighlight: runtime error: invalid memory address or nil pointer dereference

Source comment: https://github.com/microsoft/typescript-go/pull/3454#issuecomment-4285883821

```
textDocument/documentHighlight
github.com/microsoft/typescript-go/internal/lsp.(*Server).recover(0x2ac026587508, 0x2ac0365eb590)
	/mnt/vss/_work/1/s/typescript-go-3454/internal/lsp/server.go:940 +0x47
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).getReferencedSymbolsForModule(0x2ac02d517cd0?, {0x1144358?, 0x2ac03673f7a0?}, 0x2ac02744f408, 0x0, 0x0, {0x2ac02d517cd0, 0x1, 0x1}, 0x2ac02d517cd8)
	/mnt/vss/_work/1/s/typescript-go-3454/internal/ls/findallreferences.go:1312 +0x81
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).getReferencedSymbolsForNode(0x2ac02c95e700, {0x1144358, 0x2ac03673f7a0}, 0x356, 0x2ac0546a3c80, 0x2ac02744f408, {0x2ac02d517cd0, 0x1, 0x1}, {0x0, ...})
	/mnt/vss/_work/1/s/typescript-go-3454/internal/ls/findallreferences.go:916 +0x62c
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).getSemanticDocumentHighlights(0x2ac02c95e700, {0x1144358?, 0x2ac03673f7a0?}, 0x1132ec8?, 0x0?, 0x1?, {0x2ac02d517cd0, 0x1, 0xd0?})
	/mnt/vss/_work/1/s/typescript-go-3454/internal/ls/documenthighlights.go:104 +0x56
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).provideDocumentHighlightsWorker(0x2ac02c95e700, {0x1144358, 0x2ac03673f7a0}, {0x2ac04d393a40, 0x4a}, {0x2f?, 0x2?}, {0x0, 0x0, 0x0})
	/mnt/vss/_work/1/s/typescript-go-3454/internal/ls/documenthighlights.go:89 +0x793
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).ProvideDocumentHighlights(0x0?, {0x1144358?, 0x2ac03673f7a0?}, {0x2ac04d393a40, 0x4a}, {0x41355b?, 0x0?})
	/mnt/vss/_work/1/s/typescript-go-3454/internal/ls/documenthighlights.go:19 +0x4d
github.com/microsoft/typescript-go/internal/lsp.(*Server).handleDocumentHighlight(0x2ac026824008?, {0x1144358?, 0x2ac03673f7a0?}, 0x48872f?, 0x2ac02cfb2650?)
	/mnt/vss/_work/1/s/typescript-go-3454/internal/lsp/server.go:1544 +0x35
github.com/microsoft/typescript-go/internal/lsp.init.func1.registerLanguageServiceDocumentRequestHandler[...].23.1()
	/mnt/vss/_work/1/s/typescript-go-3454/internal/lsp/server.go:811 +0xa5
github.com/microsoft/typescript-go/internal/lsp.(*Server).handleRequestOrNotification.func1()
	/mnt/vss/_work/1/s/typescript-go-3454/internal/lsp/server.go:664 +0x65
github.com/microsoft/typescript-go/internal/lsp.(*Server).dispatchLoop.func3()
	/mnt/vss/_work/1/s/typescript-go-3454/internal/lsp/server.go:532 +0x2c
created by github.com/microsoft/typescript-go/internal/lsp.(*Server).dispatchLoop in goroutine 34
	/mnt/vss/_work/1/s/typescript-go-3454/internal/lsp/server.go:531 +0x531
```

<h4>Affected repos</h4>
<details>
<summary><a href="https://github.com/firecrawl/firecrawl">firecrawl/firecrawl</a></summary>
Raw error text: <code>RepoResults1/firecrawl.firecrawl.rawError.txt</code> in the <a href="https://typescript.visualstudio.com/TypeScript/_build/results?buildId=167768&view=artifacts&type=publishedArtifacts">artifact folder</a> <br />
Replay commands: <code>RepoResults1/firecrawl.firecrawl.replay.txt</code> in the <a href="https://typescript.visualstudio.com/TypeScript/_build/results?buildId=167768&view=artifacts&type=publishedArtifacts">artifact folder</a>
<h4>Old server result</h4>

```
textDocument/documentHighlight
github.com/microsoft/typescript-go/internal/lsp.(*Server).recover(0xc7020d1c308, 0xc703ad152f0)
	/mnt/vss/_work/1/s/typescript-go-main/internal/lsp/server.go:940 +0x47
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).getReferencedSymbolsForModule(0xc703498cd68?, {0x1143178?, 0xc703ac449c0?}, 0xc704798f908, 0x0, 0x0, {0xc703498cd68, 0x1, 0x1}, 0xc703498cd70)
	/mnt/vss/_work/1/s/typescript-go-main/internal/ls/findallreferences.go:1312 +0x81
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).getReferencedSymbolsForNode(0xc7048f02e00, {0x1143178, 0xc703ac449c0}, 0x356, 0xc702c658f00, 0xc704798f908, {0xc703498cd68, 0x1, 0x1}, {0x0, ...})
	/mnt/vss/_work/1/s/typescript-go-main/internal/ls/findallreferences.go:916 +0x62c
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).getSemanticDocumentHighlights(0xc7048f02e00, {0x1143178?, 0xc703ac449c0?}, 0x1131ce8?, 0x0?, 0x1?, {0xc703498cd68, 0x1, 0x83?})
	/mnt/vss/_work/1/s/typescript-go-main/internal/ls/documenthighlights.go:104 +0x56
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).provideDocumentHighlightsWorker(0xc7048f02e00, {0x1143178, 0xc703ac449c0}, {0xc7059904370, 0x4a}, {0x2f?, 0x2?}, {0x0, 0x0, 0x0})
	/mnt/vss/_work/1/s/typescript-go-main/internal/ls/documenthighlights.go:89 +0x793
github.com/microsoft/typescript-go/internal/ls.(*LanguageService).ProvideDocumentHighlights(0xc702604c5f8?, {0x1143178?, 0xc703ac449c0?}, {0xc7059904370, 0x4a}, {0x41355b?, 0x0?})
	/mnt/vss/_work/1/s/typescript-go-main/internal/ls/documenthighlights.go:19 +0x4d
github.com/microsoft/typescript-go/internal/lsp.(*Server).handleDocumentHighlight(0xc7020baf008?, {0x1143178?, 0xc703ac449c0?}, 0x48872f?, 0xc702604c650?)
	/mnt/vss/_work/1/s/typescript-go-main/internal/lsp/server.go:1544 +0x35
github.com/microsoft/typescript-go/internal/lsp.init.func1.registerLanguageServiceDocumentRequestHandler[...].23.1()
	/mnt/vss/_work/1/s/typescript-go-main/internal/lsp/server.go:811 +0xa5
github.com/microsoft/typescript-go/internal/lsp.(*Server).handleRequestOrNotification.func1()
	/mnt/vss/_work/1/s/typescript-go-main/internal/lsp/server.go:664 +0x65
github.com/microsoft/typescript-go/internal/lsp.(*Server).dispatchLoop.func3()
	/mnt/vss/_work/1/s/typescript-go-main/internal/lsp/server.go:532 +0x2c
created by github.com/microsoft/typescript-go/internal/lsp.(*Server).dispatchLoop in goroutine 36
	/mnt/vss/_work/1/s/typescript-go-main/internal/lsp/server.go:531 +0x531
```
<h4>Last few requests</h4>

```json
{"kind":"notification","method":"textDocument/didChange","params":{"textDocument":{"uri":"@PROJECT_ROOT_URI@/apps/api/src/types/parse-diff.d.ts","version":276},"contentChanges":[{"range":{"start":{"line":0,"character":30},"end":{"line":0,"character":30}},"text":" //comment"}]}}
{"kind":"notification","method":"textDocument/didChange","params":{"textDocument":{"uri":"@PROJECT_ROOT_URI@/apps/api/src/types/parse-diff.d.ts","version":277},"contentChanges":[{"range":{"start":{"line":42,"character":16},"end":{"line":42,"character":16}},"text":"Q"}]}}
{"kind":"request","method":"textDocument/codeAction","params":{"textDocument":{"uri":"@PROJECT_ROOT_URI@/apps/api/src/types/parse-diff.d.ts"},"range":{"start":{"line":43,"character":4},"end":{"line":43,"character":4}},"context":{"diagnostics":[],"only":["refactor"]}}}
{"kind":"request","method":"textDocument/documentHighlight","params":{"textDocument":{"uri":"@PROJECT_ROOT_URI@/apps/api/src/types/parse-diff.d.ts"},"position":{"line":47,"character":2}}}

```

<h4>Repro steps</h4>

```bash
#!/bin/bash

git clone https://github.com/firecrawl/firecrawl --recurse-submodules
git -C "./firecrawl" reset --hard 0ae6387b762c7450190eb7d8f9f7b81b7adfcaab
# Install packages (exact steps are below, but it might be easier to follow the repo readme)
npm --prefix "./firecrawl/examples/scrape_and_analyze_airbnb_data_e2b" ci --prefer-offline --no-audit --no-progress --legacy-peer-deps --ignore-scripts -q
pnpm --dir "./firecrawl/apps/ui/ingestion-ui" install --no-frozen-lockfile --prefer-offline --ignore-scripts --reporter=silent
pnpm --dir "./firecrawl/apps/test-suite" install --no-frozen-lockfile --prefer-offline --ignore-scripts --reporter=silent
pnpm --dir "./firecrawl/apps/test-site" install --no-frozen-lockfile --prefer-offline --ignore-scripts --reporter=silent
pnpm --dir "./firecrawl/apps/playwright-service-ts" install --no-frozen-lockfile --prefer-offline --ignore-scripts --reporter=silent
pnpm --dir "./firecrawl/apps/js-sdk" install --no-frozen-lockfile --prefer-offline --ignore-scripts --reporter=silent
pnpm --dir "./firecrawl/apps/js-sdk/firecrawl" install --no-frozen-lockfile --prefer-offline --ignore-scripts --reporter=silent
pnpm --dir "./firecrawl/apps/api" install --no-frozen-lockfile --prefer-offline --ignore-scripts --reporter=silent
downloadUrl=$(curl -s "https://typescript.visualstudio.com/TypeScript/_apis/build/builds/167768/artifacts?artifactName=RepoResults1&api-version=7.0" | jq -r ".resource.downloadUrl")
wget -O RepoResults1.zip "$downloadUrl"
unzip -p RepoResults1.zip RepoResults1/firecrawl.firecrawl.replay.txt > firecrawl.firecrawl.replay.txt
npm install --no-save @typescript/server-replay
```

To run the repro:
```bash
# `npx tsreplay --help` to learn about helpful switches for debugging, logging, etc.
npx tsreplay ./firecrawl ./firecrawl.firecrawl.replay.txt <PATH_TO_tsserver.js>
```

</details>

Additional raw reports are available in the source comment links above.
